src/main/conversions.c

```diff
@@ -969,30 +975,21 @@ as_status pyobject_to_record(AerospikeClient *self, as_error *err,
 
         while (PyDict_Next(py_rec, &pos, &key, &value)) {
 
-            if (PyUnicode_Check(key)) {
-                py_ukey = PyUnicode_AsUTF8String(key);
-                if (!py_ukey) {
-                    return as_error_update(
-                        err, AEROSPIKE_ERR_CLIENT,
-                        "Unicode bin name not encoded in utf-8.");
-                }
-                name = PyBytes_AsString(py_ukey);
-            }
-            else if (PyString_Check(key)) {
-                name = PyString_AsString(key);
-            }
-            else {
+            if (!PyUnicode_Check(key)) {
                 return as_error_update(
                     err, AEROSPIKE_ERR_CLIENT,
                     "A bin name must be a string or unicode string.");
```

Comment:
I know this is old but this error should only mention "string" now. It's implied that strings in Python 3.x are unicode.

Future Suggestion: In general we should try to remove code and messages that make this distinction. Removing macros.h would be a good step.

Reply:
I have a [PR](https://github.com/aerospike/aerospike-client-python/pull/389) to remove all the Python 2 legacy code, but Ronen decided not to move forward with it because we're switching to a new Python client written in native Python code.